tests/scripts/local/flake8_plugin_test.py

```diff
@@ -0,0 +1,87 @@
+import ast
+import unittest
+
+from scripts.local.flake8_plugin import S001, S002, SentryCheck
+
+
+def _run(src):
+    lines = src.splitlines(True)
+    tree = ast.parse(src)
+    return list(SentryCheck(filename="example.py", tree=tree, lines=lines).run())
+
+
+class SentryCheckTestCase(unittest.TestCase):
+    maxDiff = None
+
+    def errors(self, *errors):
+        return [SentryCheck.adapt_error(e) for e in errors]
+
+    def test_S001(self):
+        S001_py = """\
+class A:
+    def assert_called_once(self):
+        pass
+
+
+A().assert_called_once()
+"""
+
+        errors = _run(S001_py)
+        self.assertEqual(errors, self.errors(S001(6, 0, vars=("assert_called_once",))))
+
+    def test_S002(self):
+        S002_py = """\
+print("print statements are not allowed")
+"""
+
+        errors = _run(S002_py)
+        self.assertEqual(errors, self.errors(S002(1, 0)))
+
+    def test_S003(self):
+        S003_py = """\
+import json
+import simplejson
+from json import loads, load
+from simplejson import JSONDecoder, JSONDecodeError, _default_encoder
+import sentry.utils.json as good_json
+from sentry.utils.json import JSONDecoder, JSONDecodeError
+
+
+def bad_code():
+    a = json.loads("''")
+    b = simplejson.loads("''")
+    c = loads("''")
+    d = load()
+"""
+
+        errors = _run(S003_py)
+        assert errors == [
+            (
+                1,
+                0,
+                "S003: Use ``from sentry.utils import json`` instead.",
+                SentryCheck,
+            ),
+            (
+                2,
+                0,
+                "S003: Use ``from sentry.utils import json`` instead.",
+                SentryCheck,
+            ),
+            (
+                3,
+                0,
+                "S003: Use ``from sentry.utils import json`` instead.",
+                SentryCheck,
+            ),
+            (
+                4,
+                0,
+                "S003: Use ``from sentry.utils import json`` instead.",
+                SentryCheck,
+            ),
+        ]
+
+
+if __name__ == "__main__":
```

Comment:
I would just run this file with pytest like everything else. I am not sure if the new tests will actually already be picked up by CI. Probably not. Perhaps we don't care/don't want to run those tests on every commit anyway

Reply:
ah yeah it does run with pytest, I just copied the full file over from the old repository (pytest runs unittest tests directly)

Reply:
so what i meant with that is that I am unsure whether _this repo's_ CI setup will pick it up. we have dedicated jobs for every subfolder of `tests/`, roughly speaking

Reply:
ahhhh, I'll look into that 👍